### PR TITLE
added error handling and tests for extract.py

### PR DESCRIPTION
**Pull request by IshikaM19**
I added some Error handling on the extract script. Just checking that the arguments are of the correct type. I've also added some basic tests to test_extract.py. Please feel free to add more tests as I only added the ones I could think of. I also modified the requirements.txt in the pipeline folder...added pytest and requests_mock.

### CODE REVIEW

File: Pipeline/test_extract.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+def test_get_plants_data_not_one_value_not_an_int():
+    """Tests that checks raises an error if plant range is not a list of ints"""
+
+    plant_range = [1, 2, 3, '4']
+
+    with pytest.raises(TypeError):
+        get_plant_data(plant_range)
+
+
+def test_get_plants_data(requests_mock):
```
Comment (clv1) [inline Pipeline/test_extract.py:38]: suggested_name: `test_get_plants_data_**success**`

File: Pipeline/extract.py [hunk 2]
```diff
@@ -8,11 +8,19 @@
 
 def get_plant_data(plant_range: list[int]) -> list[dict]:
     """Gets the data from the plants api"""
+
+    if not isinstance(plant_range, list):
+        raise TypeError('The plant range must be a list of integers')
+
+    if not all(isinstance(x, int) for x in plant_range):
```
Comment (clv1) [inline Pipeline/extract.py:15]: You are raising the same error for both of these cases - could they be combined into a single if statement?

**Review by clv1 — CHANGES_REQUESTED** [review]

File: Pipeline/test_extract.py (hunk 1 above)
Comment (IshikaM19) [reply Pipeline/test_extract.py:38]: Good idea :)

File: Pipeline/extract.py (hunk 2 above)
Comment (IshikaM19) [reply Pipeline/extract.py:15]: Yea absolutely...I'll merge them together :)

**Review by clv1 — APPROVED** [review]

**Pull request merged**